src/openqaoa-core/tests/test_bpsp.py

```diff
@@ -0,0 +1,126 @@
+import unittest
+import numpy as np
+import networkx as nx
+from openqaoa.problems import BPSP
+
+
+def terms_list_equality(terms_list1, terms_list2):
+    """
+    Check the terms equality between two terms list
+    where the order of edges do not matter.
+    """
+    if len(terms_list1) != len(terms_list2):
+        bool = False
+    else:
+        for term1, term2 in zip(terms_list1, terms_list2):
+            bool = True if (term1 == term2 or term1 == term2[::-1]) else False
+
+    return bool
+
+
+class TestBPSP(unittest.TestCase):
+    """
+    Test suite for the BPSP problem.
+
+    This test suite checks the functionality of methods related to the BPSP problem,
+    including QUBO creation, car sequencing, solution methods, and more.
+    """
+
+    def setUp(self):
```

Comment:
Having this internal `self.bp` is slightly confusing considering we sometimes test using it and sometimes create a new problem.
This comment can be applied to the rest of the tests that use `self.bp`

Reply:
Do you want me to change the name of self variable, or not use a self variable at all?

Reply:
The name is fine, I'm more concerned about the readability. The issue is that some tests are performed on `self.bp`, while others are made on a new problem class. 
I think it makes more sense to create a problem class in each test and then the relevant functionality of your problem class.

Reply:
I have removed the `setUp` function and the `self.bp = BPSP(np.array([0, 1, 0, 1]))`. Wherever, I  have used `self.bp`, I am creating a new object `bpsp = BPSP(np.array([0, 1, 0, 1]))`.